x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/analysis/analyzer/Verifier.java

```diff
@@ -310,11 +310,31 @@ private static boolean checkGroupByOrder(LogicalPlan p, Set<Failure> localFailur
                 Aggregate a = (Aggregate) child;
 
                 Map<Expression, Node<?>> missing = new LinkedHashMap<>();
+
+                // track aggs and non-aggs - to keep the final modifier, use an array
+                final Expression[] aggAndNonAgg = new Expression[2];
```

Comment:
I don't think so because one would have two arrays - this is really a hack for the setting the state outside a lamba expression. I'll probably go for a `Holder` class which be more meaningful in intent.